brickstrap.sh

```diff
@@ -19,29 +20,51 @@
 # http://twitter.com/kvz
 #
 
-### Configuration
+### Runtime
 #####################################################################
+set -e
+
+# Bash will remember & return the highest exitcode in a chain of pipes.
+set -o pipefail
 
 # Environment variables
 [ -z "${LOG_LEVEL}" ] && LOG_LEVEL="3" # 4 = debug -> 0 = fail
 
-# Commandline options. This defines the usage page
+#
+# Get the brickstrap source directory
+#
+function br_script_path()
+{
+    SCRIPT_PATH=$(readlink -f "$0")
+    SCRIPT_PATH=$(dirname "$SCRIPT_PATH")
+    echo -n "$SCRIPT_PATH"
+}
 
-read -r -d '' usage <<-'EOF'
-  usage: brickstrap [-b <board>] [-d <dir>] <command>
+. "$(br_script_path)/brickstrap-components.sh"
+
+# Commandline options. This defines the usage page
+BRP_USAGE=$(cat <<'EOF'
+Usage: brickstrap -p <project> [-c <component>] [-d <dir>] <command>
```

Comment:
The idea here is only have to specify the configuration options once using one of the "creation" commands. Since the most basic creation command is `create-conf` (which currently just creates `multistrap.conf`), we need to create a `brickstrap.conf` as well. It would just need to save the absolute path to the project and each component that was passed in. All other commands source brickstrap.conf (and fail if it does not exist).